apps/store/src/features/CartDiscount/CartDiscount.tsx

```diff
@@ -0,0 +1,52 @@
+import { type FormEvent } from 'react'
+import { Discount } from '@/components/ProductCard/Discount/Discount'
+import { type ShopSession } from '@/services/shopSession/ShopSession.types'
+import { useDiscount } from './hooks/useDiscount'
+
+const FORM_CAMPAIGN_CODE = 'campaignCode'
+
+type Props = {
+  shopSession: ShopSession
+}
+
+export function CartDiscount({ shopSession }: Props) {
+  const {
+    campaignCode,
+    codeExplanation,
+    isLoading,
+    errorMessage,
+    redeemCampaign,
+    unredeemCampaign,
+  } = useDiscount(shopSession)
+
+  const redeem = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+
+    const formData = new FormData(event.currentTarget)
+    const code = formData.get(FORM_CAMPAIGN_CODE)
+
+    if (typeof code === 'string') {
+      redeemCampaign(code)
+    }
+  }
+
+  const unredeem = (event?: FormEvent<HTMLFormElement>) => {
+    if (event) {
+      event.preventDefault()
+    }
+
+    unredeemCampaign()
+  }
```

Comment:
Weird that `event` can be undefined here 🤔 

nit/personal preference:
```suggestion
  const unredeem = (event?: FormEvent<HTMLFormElement>) => {
    event?.preventDefault()

    unredeemCampaign()
  }
```

Reply:
It's because the function is passed to a form as well as invoked directly when the discount toggle is turned off!